File: fizzbuzz.cpp
#include <iostream>
#include <vector>
using namespace std;

void Fizz_Buzz(int a)
{
    int i;
    for(i=0;i<a+1;i++)
    {
        if(i ==0 || i==1)
        {
                std::cout << i  << std::endl;
        }
        else if(i%3 ==0 and i%5==0)
        {
            std::cout << "Fizz Buzz" << std::endl;
        }
        else if(i%3 ==0)
        {
            std::cout << "Fizz" << std::endl;
        }
        else if(i%5 ==0)
        {
            std::cout << "Fizz" << std::endl;
        }
        else
        {
            std::cout << i << std::endl;
        }
    }
}

int main()
{
    Fizz_Buzz(30);

}

File: Rust/fizzbuzz.rs
   pub fn fizz_buzz(n: i32) -> Vec<String> {
        let to_fizz_buzz = |x: i32| {
            if x % 15 == 0 {
                String::from("FizzBuzz")
            } else if x % 3 == 0 {
                String::from("Fizz")
            } else if x % 5 == 0 {
                String::from("Buzz")
            } else {
                x.to_string()
            }
        };
        
        (1..n + 1).map(to_fizz_buzz).collect()
    }
